pkg/controller/controller_utils.go

```diff
@@ -542,6 +544,39 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	return pod, nil
 }
 
+func (r RealPodControl) validateNodeConflict(pod *v1.Pod) error {
+	if len(pod.Spec.NodeName) == 0 {
+		return nil
+	}
+
+	node, err := r.KubeClient.Core().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nodeInfo := schedulercache.NewNodeInfo()
+	err = nodeInfo.SetNode(node)
+	if err != nil {
+		return err
+	}
+
+	fit, reasons, err := predicates.PodSelectorMatches(pod, nil, nodeInfo)
```

Comment:
Basically agree with you folks. But TBH it's really a bad experience for users to see hundreds of pods, of a RC with replicas==1,  deleting and creating. I think we should do something to avoid this.

Some other questions. If we specify nodeName for a pod, this pod will be directly bound to that node without scheduler taking effect, right? When kubelet watches this pod, it admits the pod. The [Admit](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/lifecycle/predicate.go#L52) includes `PodFitsHost` and `PodSelectorMatches`. If PredicateFailureError occurs, kubelet will reject this pod. Why do we care about scheduler rules here?

Maybe we should give `nodeName` a high priority. If users specifies `nodeName` in the spec, then nodeSelector and nodeAffinity will be ignored. [example](https://github.com/xiangpengzhao/kubernetes/commit/0b5ef7551acb76ec435a3c025140a1d060d4a155)
cc @davidopp